main.go

```diff
@@ -98,38 +133,134 @@ func getConfig(params AppConfigParams) ([]string, error) {
 		return nil, err
 	}
 
-	vars := getVars(string(configuration.Configuration))
+	return configuration.Configuration, nil
+}
+
+// getVars parses a config in env format and returns a slice of variable-value strings like "VAR=value" suitable to
+// supply to the Env attribute of the os/exec Cmd struct.
+func getVars(config []byte) ([]string, error) {
+	vars, err := godotenv.Parse(bytes.NewReader(config))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse configuration from AppConfig: %w", err)
+	}
 
 	fmt.Printf("read %d lines from AppConfig\n", len(vars))
 	if verbose {
 		fmt.Printf("vars: %s\n", vars)
 	}
 
-	return vars, nil
+	varSlice := make([]string, 0, len(vars))
+	for k, v := range vars {
+		varSlice = append(varSlice, k+"="+v)
```

Comment:
The lack of proper quoting or escaping of the value here seems like it might cause problems at some point (such as perhaps if the value contains a `#`, `"`, or `=`, though those are just guesses at special characters for the env. var. context).

Goes godotenv provide any helper functions for setting/escaping/encoding environment variables?

Reply:
Actually, this isn't being saved to a file. It's being sent to the subprocess, which doesn't parse any kind of escape characters or quoting. Everything is taken literally. Just as when you run `env` in a bash shell. The function doc tries to describe this a bit.